swoole_http_client_coro.cc

```diff
@@ -1404,6 +1413,30 @@ bool http_client::close()
     return false;
 }
 
+std::string http_client::get_basic_auth_encode(char *name, size_t nameLen, char *passwd, size_t passwdLen){
```

Comment:
1. code style (use snake name, put the brace on a new line)
2. keep same with PHP API (`set_basic_auth`) or let it be a static function
3. use `emalloc` (this module has already been strongly associated with PHP)
4. why not using `sw_sprintf`?